admin/install/install.go

```diff
@@ -0,0 +1,121 @@
+// Copyright 2017 Google Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License"); you may not
+// use this file except in compliance with the License.  You may obtain a copy
+// of the License at:
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+// WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.  See the
+// License for the specific language governing permissions and limitations
+// under the License.
+
+// The install binary installs sandboxfs and all of its support files.
+package main
+
+import (
+	"flag"
+	"fmt"
+	"os"
+	"path/filepath"
+
+	"github.com/bazelbuild/sandboxfs/internal/shell"
+)
+
+const (
+	// Relative path to the runfiles directory for the install script.  Used if the script is
+	// run by hand from the workspace directory, without using "bazel run".
+	runfiles = "bazel-bin/admin/install/install.runfiles/sandboxfs"
+
+	// Path to the built sandboxfs binary relative to the runfiles directory.
+	sandboxfsBin = "cmd/sandboxfs/sandboxfs"
+
+	// Desired permissions for installed files.
+	dataPerm = 0644
+	execPerm = 0755
+	dirPerm  = 0755
+)
+
+// installSpec contains the details for the installation of one source file.
+type installSpec struct {
+	// sourceFile contains the relative path to the source file to be installed.  If ran with
+	// Bazel, all source files must be present in the runfiles directory of this tool, which
+	// means they all must be listed as data dependencies.
+	sourceFile string
+
+	// targetDir indicates the relative path from the prefix where the file will be installed.
+	// The name of the installed file matches the name of the source file.
+	targetDir string
+
+	// mode indicates the desired permissions of the installed file.
+	mode os.FileMode
+}
+
+// installData specifies the files to be installed by this tool.
+var installData = []installSpec{
```

Comment:
Does installData need to be global?

Reply:
It doesn't *have* to be, but I considered this to be the program's data and thus felt better represented as a constant. (It cannot be a constant, though, because constants in Go can only be of primitive types.)